Move the template mode edit and new links to a dedicated template panel

youknowriad opened the pull request:
Right now, we have two places where we can view and edit the template of a page/post. The historical "page attributes"  panel and the new "template" row under "Post status". This PR consolidates under the "Page/Post attributes" panel.

There are on going explorations to add flows related to the templates to the top toolbar (title) but this should be added separately.

That said I do wonder whether it makes sense to extract this  from the "page attributes" panel into its own dedicated panel "template" right after "Post status". Did you consider that @jasmussen @jameskoster

youknowriad committed:
Move the template mode edit and new links to the post attributes panel

ntsekouras commented on packages/e2e-tests/specs/experiments/post-editor-template-mode.test.js:
```diff
@@ -9,6 +9,7 @@ import {
 	trashAllPosts,
 	openPreviewPage,
 	openDocumentSettingsSidebar,
+	openSidebarPanelWithTitle,
```
This is not an exported util.

ntsekouras reviewed (approved):
Code wise this looks good. If no other changes design wise occur, I'm pre approving. Thanks!

Left a comment just to fix the tests.

jameskoster commented:
> There are on going explorations to add flows related to the templates to the top toolbar (title) but this should be added separately.

> That said I do wonder whether it makes sense to extract this from the "page attributes" panel into its own dedicated panel "template" right after "Post status". Did you consider that @jasmussen @jameskoster

My feeling is that the answer to this question should be informed by the design solution for the top bar concept. The visual design may not be identical, but the iA should probably match as closely as possible. 

Those designs are forthcoming, but due to the volume of information we might need to display it seems fairly safe to assume that these settings will be split in to their own sections some how. 

All that to say – yes I believe a dedicated "Template" section in the Inspector is likely warranted.

jasmussen commented on packages/e2e-tests/specs/experiments/post-editor-template-mode.test.js:
```diff
@@ -42,6 +43,7 @@ describe( 'Post Editor Template mode', () => {
 
 		// Switch to template mode.
 		await openDocumentSettingsSidebar();
+		await openSidebarPanelWithTitle( 'Post Attributes' );
```
Panels should be sentence case, i.e. "Post attributes". But it looks from the code that you're just opening an existing panel, and not introducing this here. So I guess it's a separate thing.

jasmussen commented:
I was there when we added it to the status panel, but I also think it's right to move it out from there to reduce the noise, and longer term move it to the top toolbar as discussed. So moving it here makes sense:

<img width="432" alt="Screenshot 2021-04-16 at 12 51 24" src="https://user-images.githubusercontent.com/1204802/115014415-97053400-9eb2-11eb-9c6f-c2ab41616ad6.png">

But perhaps it should be a secondary button, not a link button. To Jay's point, there's a bit of design work we need to do here, so there are some incoming design changes to this. But in the mean time, just a secondary button style might be a good start.

> All that to say – yes I believe a dedicated "Template" section in the Inspector is likely warranted.

Maybe. But if that section, for now, will only contain "New" and "Edit", to me it seems fine to keep it in this attributes panel no? If it grows in time we can move it, perhaps. Not a strong opinion.

Thanks for the PR!

youknowriad commented:
> Maybe. But if that section, for now, will only contain "New" and "Edit", to me it seems fine to keep it in this attributes panel no? If it grows in time we can move it, perhaps. Not a strong opinion.

I was thinking of moving the template selector there as well. the idea for me was  just to give this a bit more visibility. It's more important than say "parent page or order"

jameskoster commented:
> Maybe. But if that section, for now, will only contain "New" and "Edit", to me it seems fine to keep it in this attributes panel no? If it grows in time we can move it, perhaps. Not a strong opinion.

It will also contain the selector, no?

My train of thought was that the attributes panel could include the template details, post title input (for when/if we remove that from the canvas) and the permalink input. But we already have a dedicated permalink panel so there's no sense duplicating it. Medium term it will probably move to the top bar along with the title input anyway. At that point the attributes panel only contains template information. Ergo, we can probably just rename it "Template".

carolinan commented:
Short term, until there is an updated template selector, it makes sense to group the template information and the page attributes template selection together.

But with this PR, I can not see which template is being used? This was previously shown next to the edit link.
If I open a new page, I would expect the template to be page, but the select list shows "Default".

youknowriad committed:
Use a dedicated panel for the post template

youknowriad committed:
Add template title to the panel title

youknowriad commented:
I've updated this PR to use a dedicated panel, I think this works way better than what we have currently (or the previous implementation of this PR). 

Also, to address @carolinan's feedback, I added the template title to the panel title when available.

youknowriad committed:
Fix e2e tests
